### PR TITLE
Parser: Fix function resolution rule for 'TIMESTAMPADD' and 'TIMESTAMPDIFF'

### PR DESCRIPTION
**Pull request by spongedu**
Built-in function 'TIMESTAMPADD' and 'TIMESTAMDIFF' don't accept the following options as time unit: 
- "SECOND_MICROSECOND"
- "MINUTE_MICROSECOND"
- "MINUTE_SECOND"
- "HOUR_MICROSECOND"
- "HOUR_SECOND"
- "HOUR_MINUTE"
- "DAY_MICROSECOND"
- "DAY_SECOND"
- "DAY_MINUTE"

"DAY_HOUR"
"YEAR_MONTH"

### CODE REVIEW

**Comment by shenli** [comment]
@spongedu Thanks for your PR!
Could you add some test cases in parser_test.go?

**Comment by shenli** [comment]
LGTM
@zimulala @tiancaiamao PTAL

File: parser/parser.y [hunk 1]
```diff
@@ -796,7 +796,8 @@ import (
 	RegexpSym		"REGEXP or RLIKE"
 	IntoOpt			"INTO or EmptyString"
 	ValueSym		"Value or Values"
-	TimeUnit		"Time unit"
+	TimeUnit		"Time unit for 'DATE_ADD', 'DATE_SUB', 'ADDDATE', 'SUBDATE', 'EXTRACT'"
+	TimeUnit2		"Time unit for 'TIMESTAMPADD' and 'TIMESTAMPDIFF'"
```
Comment (zimulala) [inline parser/parser.y]: I think `TimestampUnit` is more readable.

Comment (spongedu) [inline parser/parser.y]: OK, I'll fix.

**Comment by zimulala** [comment]
LGTM

**Pull request merged**